scripts/control_bar.js

```diff
@@ -2,21 +2,121 @@
 /* exported makeImageWidget */
 
 // Construct the image sizing controls.
-var makeImageControl = function(imageElement) {
+var makeImageControl = function(canvas, reSize) {
     let imageKey;
-    let percent;
-    const percentInput = $("<input>", {type: 'number', min: '1', max: '999', value: percent, title: texts.zoomPercent});
+    let scale = 1;
+    let imageWidth, imageHeight;
+    const image = document.createElement("img");
+    let ctx = canvas.getContext("2d");
+
+    const percentInput = $("<input>", {type: 'number', min: '1', max: '999', value: 100, title: texts.zoomPercent});
+
+    let sine = 0, cosine = 1;
+
+    let scrollbarWidth = false;
+    var getScrollbarWidth = function() {
+        if (scrollbarWidth === false) {
+            let testDiv = document.createElement('div');
+            testDiv.innerHTML = '<div style="width:50px;height:5px;position:absolute; overflow:scroll;"><div style="width:100px;height:5px;"></div></div>';
+            let innerDiv = testDiv.firstChild;
+            document.body.appendChild(testDiv);
+            scrollbarWidth = innerDiv.offsetWidth - innerDiv.clientWidth;
+            document.body.removeChild(testDiv);
+        }
+        return scrollbarWidth;
+    };
 
-    function setZoom() {
-        imageElement.width(10 * percent);
-        imageElement.height("auto");
+    function maxVal(a, b) {
+        return (a > b) ? a : b;
     }
 
-    function saveZoom() {
+    // If we always make the canvas fit image then when reducing the image size
+    // rapidly with spinner the canvas does not get cleared even when
+    // redrawing with setTimer(0).
+    // So, make canvas dimension fit image dimension only if it is larger
+    // than the pane. A scroll bar will then appear. Otherwise make canvas
+    // dimension fill pane. Canvas now always clears.
+    // This introduces a new problem: if the image is smaller than the pane
+    // and the pane size decreases it is possible to scroll the image out
+    // of view. So redraw when pane size changes.
+    // This would be easy if we could use ResizeObserver(). In the meantime
+    // use a resize callback which gets fired when pane size changes.
+    function reDraw() {
+        getScrollbarWidth();
+        // clearRect acts through transform
+        ctx.resetTransform();
+        ctx.clearRect( 0, 0, canvas.width, canvas.height);
+        let scaleWidth = imageWidth * scale;
+        let scaleHeight = imageHeight * scale;
+        let newWidth, newHeight;
+        if(cosine != 0) {
```

Comment:
same here with the if.

Reply:
I agree it looks better with a space but they are all ike this